helix-term/src/ui/mod.rs

```diff
@@ -29,6 +29,7 @@ pub fn regex_prompt(
     cx: &mut crate::commands::Context,
     prompt: String,
     fun: impl Fn(&mut View, &mut Document, &mut Registers, Regex) + 'static,
+    on_validate: impl Fn(&mut View, Regex) + 'static,
```

Comment:
Let's use the same function signature for both functions (note that this changed on master and it no longer requires `&mut Registers`: `fun: impl Fn(&mut View, &mut Document, Regex) + 'static,`

Reply:
Actually, let's unify the two. `fun: impl Fn(&mut View, &mut Document, Regex, PromptEvent) + 'static`, call the same function on both Validate and Update and pass through the event. you can just

```rust
if event != PromptEvent::Validate {
    return
}
```

in all the old methods (keep_selections, etc)